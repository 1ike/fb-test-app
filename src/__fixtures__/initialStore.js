const options = {
  preset: "islands#blueCircleDotIconWithCaption",
  draggable: true
};

const point1 = {
  id: 0,
  name: "p0",
  placemark: {
    geometry: {
      coordinates: [55.78, 37.63]
    },
    properties: {
      hintContent: "p0",
      balloonContent: "p0"
    },
    options
  }
};
const point2 = {
  id: 1,
  name: "p1",
  placemark: {
    geometry: {
      coordinates: [55.74, 37.62]
    },
    properties: {
      hintContent: "p1",
      balloonContent: "p1"
    },
    options
  }
};
const point3 = {
  id: 2,
  name: "p2",
  placemark: {
    geometry: {
      coordinates: [55.76, 37.7]
    },
    properties: {
      hintContent: "p2",
      balloonContent: "p2"
    },
    options
  }
};

export default {
  points: [point1, point2, point3]
};

export const reorderedPoints = {
  points: [
    JSON.parse(JSON.stringify(point2)),
    JSON.parse(JSON.stringify(point1)),
    JSON.parse(JSON.stringify(point3))
  ]
};
