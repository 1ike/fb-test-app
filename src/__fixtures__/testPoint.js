export const name = "test point 1";
export const coordinates = [50, 70];
export const point = {
  id: 5,
  name,
  placemark: {
    geometry: {
      coordinates
    },
    properties: {
      hintContent: name,
      balloonContent: name
    },
    options: {
      preset: "islands#blueCircleDotIconWithCaption",
      draggable: true
    }
  }
};
