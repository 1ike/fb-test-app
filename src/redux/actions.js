import { createActions } from "redux-actions";

export const {
  createPoint,
  deletePoint,
  shiftPoint,
  reoderPoints,
  shiftCenter
} = createActions(
  // {
  //   ACTION_ONE: (key, value) => ({ [key]: value })
  // },
  "CREATE_POINT",
  "DELETE_POINT",
  "SHIFT_POINT",
  "REODER_POINTS",
  "SHIFT_CENTER"
);
