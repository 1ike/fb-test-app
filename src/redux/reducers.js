import { combineReducers } from "redux";
import { reducer as form } from "redux-form";
import { handleActions } from "redux-actions";

import * as actions from "./actions";

const points = handleActions(
  {
    [actions.createPoint](state, { payload: point }) {
      return [...state, point];
    },
    [actions.deletePoint](state, { payload: id }) {
      return state.filter(point => point.id !== id);
    },
    [actions.shiftPoint](
      state,
      {
        payload: { id, coordinates }
      }
    ) {
      return state.map(point => {
        if (point.id !== id) return point;
        point.placemark.geometry.coordinates = coordinates;
        return point;
      });
    },
    [actions.reoderPoints](state, { payload: points }) {
      return points;
    }
  },
  []
);

const center = handleActions(
  {
    [actions.shiftCenter](state, { payload }) {
      return payload;
    }
  },
  [55.76, 37.64]
);

export default combineReducers({
  form,
  points,
  center
});
