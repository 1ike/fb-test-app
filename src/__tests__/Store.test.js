import { createStore } from "redux";

import reducers from "../redux/reducers";
import {
  createPoint,
  deletePoint,
  shiftPoint,
  reoderPoints,
  shiftCenter
} from "../redux/actions";

import initialStore, { reorderedPoints } from "../__fixtures__/initialStore";
import { point, coordinates } from "../__fixtures__/testPoint";

describe("Redux store", () => {
  let store;

  beforeEach(() => {
    store = createStore(reducers, initialStore);
  });

  it("createPoint", () => {
    store.dispatch(createPoint(point));
    expect(store.getState()).toMatchSnapshot();
  });

  it("deletePoint", () => {
    store.dispatch(deletePoint(1));
    expect(store.getState()).toMatchSnapshot();
  });

  it("shiftPoint", () => {
    store.dispatch(shiftPoint({ id: 0, coordinates }));
    expect(store.getState()).toMatchSnapshot();
  });

  it("reoderPoints", () => {
    store.dispatch(reoderPoints(reorderedPoints));
    expect(store.getState()).toMatchSnapshot();
  });

  it("shiftCenter", () => {
    store.dispatch(shiftCenter(coordinates));
    expect(store.getState()).toMatchSnapshot();
  });
});
