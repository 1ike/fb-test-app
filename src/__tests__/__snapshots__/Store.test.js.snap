// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Redux store createPoint 1`] = `
Object {
  "center": Array [
    55.76,
    37.64,
  ],
  "form": Object {},
  "points": Array [
    Object {
      "id": 0,
      "name": "p0",
      "placemark": Object {
        "geometry": Object {
          "coordinates": Array [
            55.78,
            37.63,
          ],
        },
        "options": Object {
          "draggable": true,
          "preset": "islands#blueCircleDotIconWithCaption",
        },
        "properties": Object {
          "balloonContent": "p0",
          "hintContent": "p0",
        },
      },
    },
    Object {
      "id": 1,
      "name": "p1",
      "placemark": Object {
        "geometry": Object {
          "coordinates": Array [
            55.74,
            37.62,
          ],
        },
        "options": Object {
          "draggable": true,
          "preset": "islands#blueCircleDotIconWithCaption",
        },
        "properties": Object {
          "balloonContent": "p1",
          "hintContent": "p1",
        },
      },
    },
    Object {
      "id": 2,
      "name": "p2",
      "placemark": Object {
        "geometry": Object {
          "coordinates": Array [
            55.76,
            37.7,
          ],
        },
        "options": Object {
          "draggable": true,
          "preset": "islands#blueCircleDotIconWithCaption",
        },
        "properties": Object {
          "balloonContent": "p2",
          "hintContent": "p2",
        },
      },
    },
    Object {
      "id": 5,
      "name": "test point 1",
      "placemark": Object {
        "geometry": Object {
          "coordinates": Array [
            50,
            70,
          ],
        },
        "options": Object {
          "draggable": true,
          "preset": "islands#blueCircleDotIconWithCaption",
        },
        "properties": Object {
          "balloonContent": "test point 1",
          "hintContent": "test point 1",
        },
      },
    },
  ],
}
`;

exports[`Redux store deletePoint 1`] = `
Object {
  "center": Array [
    55.76,
    37.64,
  ],
  "form": Object {},
  "points": Array [
    Object {
      "id": 0,
      "name": "p0",
      "placemark": Object {
        "geometry": Object {
          "coordinates": Array [
            55.78,
            37.63,
          ],
        },
        "options": Object {
          "draggable": true,
          "preset": "islands#blueCircleDotIconWithCaption",
        },
        "properties": Object {
          "balloonContent": "p0",
          "hintContent": "p0",
        },
      },
    },
    Object {
      "id": 2,
      "name": "p2",
      "placemark": Object {
        "geometry": Object {
          "coordinates": Array [
            55.76,
            37.7,
          ],
        },
        "options": Object {
          "draggable": true,
          "preset": "islands#blueCircleDotIconWithCaption",
        },
        "properties": Object {
          "balloonContent": "p2",
          "hintContent": "p2",
        },
      },
    },
  ],
}
`;

exports[`Redux store reoderPoints 1`] = `
Object {
  "center": Array [
    55.76,
    37.64,
  ],
  "form": Object {},
  "points": Object {
    "points": Array [
      Object {
        "id": 1,
        "name": "p1",
        "placemark": Object {
          "geometry": Object {
            "coordinates": Array [
              55.74,
              37.62,
            ],
          },
          "options": Object {
            "draggable": true,
            "preset": "islands#blueCircleDotIconWithCaption",
          },
          "properties": Object {
            "balloonContent": "p1",
            "hintContent": "p1",
          },
        },
      },
      Object {
        "id": 0,
        "name": "p0",
        "placemark": Object {
          "geometry": Object {
            "coordinates": Array [
              55.78,
              37.63,
            ],
          },
          "options": Object {
            "draggable": true,
            "preset": "islands#blueCircleDotIconWithCaption",
          },
          "properties": Object {
            "balloonContent": "p0",
            "hintContent": "p0",
          },
        },
      },
      Object {
        "id": 2,
        "name": "p2",
        "placemark": Object {
          "geometry": Object {
            "coordinates": Array [
              55.76,
              37.7,
            ],
          },
          "options": Object {
            "draggable": true,
            "preset": "islands#blueCircleDotIconWithCaption",
          },
          "properties": Object {
            "balloonContent": "p2",
            "hintContent": "p2",
          },
        },
      },
    ],
  },
}
`;

exports[`Redux store shiftCenter 1`] = `
Object {
  "center": Array [
    50,
    70,
  ],
  "form": Object {},
  "points": Array [
    Object {
      "id": 0,
      "name": "p0",
      "placemark": Object {
        "geometry": Object {
          "coordinates": Array [
            50,
            70,
          ],
        },
        "options": Object {
          "draggable": true,
          "preset": "islands#blueCircleDotIconWithCaption",
        },
        "properties": Object {
          "balloonContent": "p0",
          "hintContent": "p0",
        },
      },
    },
    Object {
      "id": 1,
      "name": "p1",
      "placemark": Object {
        "geometry": Object {
          "coordinates": Array [
            55.74,
            37.62,
          ],
        },
        "options": Object {
          "draggable": true,
          "preset": "islands#blueCircleDotIconWithCaption",
        },
        "properties": Object {
          "balloonContent": "p1",
          "hintContent": "p1",
        },
      },
    },
    Object {
      "id": 2,
      "name": "p2",
      "placemark": Object {
        "geometry": Object {
          "coordinates": Array [
            55.76,
            37.7,
          ],
        },
        "options": Object {
          "draggable": true,
          "preset": "islands#blueCircleDotIconWithCaption",
        },
        "properties": Object {
          "balloonContent": "p2",
          "hintContent": "p2",
        },
      },
    },
  ],
}
`;

exports[`Redux store shiftPoint 1`] = `
Object {
  "center": Array [
    55.76,
    37.64,
  ],
  "form": Object {},
  "points": Array [
    Object {
      "id": 0,
      "name": "p0",
      "placemark": Object {
        "geometry": Object {
          "coordinates": Array [
            50,
            70,
          ],
        },
        "options": Object {
          "draggable": true,
          "preset": "islands#blueCircleDotIconWithCaption",
        },
        "properties": Object {
          "balloonContent": "p0",
          "hintContent": "p0",
        },
      },
    },
    Object {
      "id": 1,
      "name": "p1",
      "placemark": Object {
        "geometry": Object {
          "coordinates": Array [
            55.74,
            37.62,
          ],
        },
        "options": Object {
          "draggable": true,
          "preset": "islands#blueCircleDotIconWithCaption",
        },
        "properties": Object {
          "balloonContent": "p1",
          "hintContent": "p1",
        },
      },
    },
    Object {
      "id": 2,
      "name": "p2",
      "placemark": Object {
        "geometry": Object {
          "coordinates": Array [
            55.76,
            37.7,
          ],
        },
        "options": Object {
          "draggable": true,
          "preset": "islands#blueCircleDotIconWithCaption",
        },
        "properties": Object {
          "balloonContent": "p2",
          "hintContent": "p2",
        },
      },
    },
  ],
}
`;
